### PR TITLE
feat: Remove unnecessary Validate methods in connector (and gofumpt)

### PR DESCRIPTION
**Pull request by jameshoulahan**

### CODE REVIEW

File: connector/dummy.go [hunk 1]
```diff
@@ -150,10 +108,16 @@ func (conn *Dummy) CreateLabel(ctx context.Context, name []string) (imap.Mailbox
 	return mbox, nil
 }
 
-func (conn *Dummy) UpdateLabel(ctx context.Context, labelID imap.LabelID, name []string) error {
-	conn.state.updateLabel(labelID, name)
+func (conn *Dummy) UpdateLabel(ctx context.Context, labelID imap.LabelID, newName []string) error {
+	oldName := conn.state.getLabel(labelID).Name
```
Comment (LBeernaertProton) [inline connector/dummy.go:112]: missing error handling for invalid label id?

Comment (jameshoulahan) [reply connector/dummy.go:112]: eh it's just the dummy -- if we're passing in missing label IDs it's probably worth just panicking cos something's seriously wrong

Comment (LBeernaertProton) [reply connector/dummy.go:112]: But then you can't verify the rest of your code handles this gracefully.

Comment (jameshoulahan) [reply connector/dummy.go]: Fair enough, added some basic error handling.

**Review by LBeernaertProton — DISMISSED** [review]

**Review by LBeernaertProton — APPROVED** [review]

**Pull request merged**